qemu/tests/qmp_command.py

```diff
@@ -251,29 +215,45 @@ def send_cmd(cmd):
     except Exception as err:
         test.fail(err)
 
+    # sleep 10s to make device_del take effect
+    if qmp_cmd == "device_del id=device_del01; netdev_del id=netdev_del01":
+        time.sleep(10)
+
     # Post command
     if post_cmd is not None:
         logging.info("Run post command '%s'.", post_cmd)
         post_o = send_cmd(post_cmd)
+        if not isinstance(post_o, list):
+            post_o = [post_o]
         logging.debug("Post-command: '%s'\n Output: '%s'", post_cmd, post_o)
 
-    if result_check is not None:
-        txt = "Verify that qmp command '%s' works as designed." % qmp_cmd
-        logging.info(txt)
-        if result_check == "equal" or result_check == "contain":
-            if qmp_cmd == "query-name":
-                vm_name = params["main_vm"]
-                check_result(output, vm_name, exception_list)
-            elif qmp_cmd == "query-uuid":
-                uuid_input = params["uuid"]
-                check_result(output, uuid_input, exception_list)
-            else:
-                check_result(output, cmd_return_value, exception_list)
-        elif result_check == "m_format_q":
-            check_result(output, cmd_return_value, exception_list)
-        elif 'post' in result_check:
-            result_check = result_check.split('_', 1)[1]
-            check_result(post_o, cmd_return_value, exception_list)
-        else:
-            check_result(output, post_o, exception_list)
+    if result_check == "equal" or result_check == "contain":
+        logging.info("Verify qmp command '%s' works as designed." % qmp_cmd)
+        if qmp_cmd == "query-name":
+            vm_name = params["main_vm"]
+            expect_o = [{'name': vm_name}]
+        elif qmp_cmd == "query-uuid":
+            uuid_input = params["uuid"]
+            expect_o = [{'UUID': uuid_input}]
+        elif qmp_cmd == "query-version":
+            qemu_version_cmd = "rpm -qa | grep -E 'qemu-kvm(-(rhev|ma))?-[0-9]' | head -n 1"
+            host_arch = platform.machine()
+            qemu_version = callback["host_cmd"](qemu_version_cmd).replace('.%s' % host_arch, '')
+            expect_o = [{'package': qemu_version}]
+        elif qmp_cmd == "query-block":
+            images = params['images'].split()
+            image_info = {}
+            for image in images:
+                image_params = params.object_params(image)
+                image_format = image_params['image_format']
+                image_drive = "drive_%s" % image
+                image_info['device'] = image_drive
+                image_info['qdev'] = image
+                image_info['format'] = image_format
+                expect_o.append(image_info)
+        check_result(qmp_o, expect_o, exception_list)
+    elif 'post' in result_check:
+        logging.info("Verify post qmp command '%s' works as designed." % post_cmd)
+        result_check = result_check.split('_', 1)[1]
```

Comment:
`result_check = result_check.lstrip("post_")`  might be clearer.

Reply:
I refer to use `split`, since `lstrip("post_")` also can work as following, it is not what we want.
```
>>> str1 = 'p_not_contain'
>>> str1.lstrip("post_")
'not_contain'
>>> str2 = 'o_not_contain'
>>> str2.lstrip("post_")
'not_contain'
```